lib/puppet/util/splunk_hec.rb

```diff
@@ -45,10 +45,17 @@ def create_http(source_type)
 
   def submit_request(body)
     # we want users to be able to provide different tokens per sourcetype if they want
-    source_type = body['sourcetype'].split(':')[1]
-    token_name = "token_#{source_type}"
-    http = create_http(source_type)
-    token = settings[token_name] || settings['token'] || raise(Puppet::Error, 'Must provide token parameter to splunk class')
+    if body['sourcetype']
+      source_type = body['sourcetype'].split(':')[1]
+      token_name = "token_#{source_type}"
+      http = create_http(source_type)
+      token = settings[token_name] || settings['token'] || raise(Puppet::Error, 'Must provide token parameter to splunk class')
+    else
+      # For testing purposes there isn't always a sourcetype.
```

Comment:
Seems weird that this is only here to get the tests to pass. Is it an important precondition of this module that the `sourcetype` must exist? If so, may be worth finding a way to get that test to pass _without_ having this fix. That way can be done in a follow-up PR though.

Reply:
Splunk should be able to receive data without a specific sourcetype although the module isn't currently set up to do this automatically. It's nice to be able to do this for testing purposes, however an alternative approach would be to specify a sourcetype which does not exist, which creates the same behavior. That would be a way to keep this test, without modifying the production code.

Reply:
Yeah, let's do the "nonexistent sourcetype" option then if it's easy enough.

Reply:
Cool, updated!